OpenChange/iCalEvent+MAPIStore.m

```diff
@@ -371,106 +472,24 @@ - (void) updateFromMAPIProperties: (NSDictionary *) properties
   value = [properties objectForKey: @"recipients"];
   if (value)
     {
-      NSArray *recipients;
       NSDictionary *dict;
-      NSString *orgEmail, *sentBy, *attEmail;
+      NSString *orgEmail, *sentBy;
       iCalPerson *person;
       iCalPersonPartStat newPartStat;
-      NSNumber *flags, *trackStatus;
-      int i, effective;
+      int effective;
       BOOL organizerIsSet = NO;
 
       [self setOrganizer: nil];
       [self removeAllAttendees];
 
-      recipients = [value objectForKey: @"to"];
-      effective = 0;
-      for (i = 0; i < [recipients count]; i++)
-        {
-          dict = [recipients objectAtIndex: i];
-          person = [iCalPerson new];
-          [person setCn: [dict objectForKey: @"fullName"]];
-          attEmail = [dict objectForKey: @"email"];
-          [person setEmail: attEmail];
- 
-          flags = [dict objectForKey: MAPIPropertyKey (PR_RECIPIENT_FLAGS)];
-          if (!flags)
-            {
-              [self logWithFormat:
-                      @"no recipient flags specified: skipping recipient"];
-              continue;
-            }
+      effective = [self _updateFromAttendeeMAPIProperties: [value objectForKey: @"to"]
+                                                 withRole: @"REQ-PARTICIPANT"
+                                                 outParam: &organizerIsSet];
+      effective += [self _updateFromAttendeeMAPIProperties: [value objectForKey: @"cc"]
+                                                 withRole: @"OPT-PARTICIPANT"
+                                                 outParam: &organizerIsSet];
```

Comment:
Thanks for splitting the function @jacalvo :+1: 
now even I can understand what is going on easily :)